### PR TITLE
Updates

### PR DESCRIPTION
**Pull request by hkbertoson**
- Simplify the toast update logic by using `findIndex` instead of `map`
- Remove unnecessary batching and flushing code
- Improve readability and maintainability of the code
- Updated offset to improve code

### CODE REVIEW

File: src/index.tsx [hunk 1]
```diff
@@ -433,25 +435,19 @@ const Toaster = (props: ToasterProps) => {
...
+          return [
+            ...toasts.slice(0, indexOfExistingToast),
+            { ...toasts[indexOfExistingToast], ...toast },
+            ...toasts.slice(indexOfExistingToast + 1),
+          ];
+        }
 
-      // Prevent batching, temp solution.
-      setTimeout(() => {
-        ReactDOM.flushSync(() => {
```
Comment (emilkowalski) [inline src/index.tsx:439]: This is necessary to ensure correct calculation when multiple toasts are added at the same time.

Comment (hkbertoson) [reply src/index.tsx:439]: I tested it out and was able to render multiple toasts at the same time with no issues.

Comment (hkbertoson) [reply src/index.tsx:439]:
````
toast('My action toast', {
    action: {
		label: 'Action',
		onClick: () => console.log('Action!'),
	},
});
````


Just threw this inside of a button and spammed it

Comment (emilkowalski) [reply src/index.tsx:439]: Yea that’s not what I meant, you need to invoke 2 toasts with one click to experience the bug ‘flushSync’ solves

Comment (hkbertoson) [reply src/index.tsx:439]: Oh I see it now. Interesting... Okay. I will fix that back.

Comment (hkbertoson) [reply src/index.tsx:439]: I got it updated. I also added another testing button for rendering multiple toasts.

**Comment by emilkowalski** [comment]
These updates are not necessary, but thanks for the contribution 🙏🏻

**Pull request closed without merging**